playground/frontend/playground_components/tools/extract_symbols_java/src/main/java/com/playground/extract_symbols/Main.java

```diff
@@ -0,0 +1,120 @@
+        String yamlString = buildYamlString(classInfoMap);
+        System.out.println(yamlString);
+    }
+
+    private static HashMap<String, ClassInfo> getDirSymbols(String sdkPathString) throws IOException {
+        var classInfoMap = new HashMap<String, ClassInfo>();
+        var sdkPath = new File(sdkPathString).toPath().toAbsolutePath();
+        Files.walk(sdkPath).forEach(path -> {
+            var stringPath = path.toString();
+            if (stringPath.endsWith(".java") && !stringPath.contains("test")) {
+                var fileName = stringPath.substring(stringPath.lastIndexOf("/") + 1).replace(".java", "");
+                try {
+                    var unit = StaticJavaParser.parse(path);
+                    if (unit.getClassByName(fileName).isPresent()) {
+                        addClassSymbols(classInfoMap, unit.getClassByName(fileName).get());
+                    }
+                } catch (IOException | ParseProblemException ignored) {
+                }
+            }
+        });
+
+        return classInfoMap;
+    }
+
+    private static void addClassSymbols(HashMap<String, ClassInfo> classInfoList, ClassOrInterfaceDeclaration cl) {
+        if (!cl.isPublic()) {
+            return;
+        }
+
+        ClassInfo classInfo;
+        if (classInfoList.containsKey(cl.getNameAsString())) {
+            classInfo = classInfoList.get(cl.getNameAsString());
+        } else {
+            classInfo = new ClassInfo();
+            classInfoList.put(cl.getNameAsString(), classInfo);
+        }
+
+        cl.findAll(MethodDeclaration.class).forEach(method -> {
+            if (method.isPublic()) {
+                classInfo.publicMethods.add(method.getNameAsString());
+            }
+        });
+        cl.findAll(FieldDeclaration.class).forEach(field -> {
+            if (field.isPublic()) {
+                classInfo.publicFields.add(field.getVariable(0).getNameAsString());
+            }
+        });
+    }
+
+    private static String buildYamlString(HashMap<String, ClassInfo> classInfoMap) throws YamlException {
+        var stringWriter = new StringWriter();
+        var yamlWriter = new YamlWriter(stringWriter);
+        yamlWriter.getConfig().writeConfig.setWriteClassname(YamlConfig.WriteClassName.NEVER);
+        var yamlMap = new LinkedHashMap<String, Map<String, List<String>>>();
+
+        classInfoMap.forEach((key, value) -> yamlMap.put(key, value.toMap()));
+        var sortedMap = sortMap(yamlMap);
+
+        yamlWriter.write(sortedMap);
+
+        yamlWriter.close();
+        return stringWriter.toString();
+    }
+
+    private static LinkedHashMap<String, Map<String, List<String>>> sortMap(HashMap<String, Map<String, List<String>>> yamlMap) {
+        var comparator = new Comparator<Map.Entry<String, Map<String, List<String>>>>() {
+            @Override
+            public int compare(Map.Entry<String, Map<String, List<String>>> o1, Map.Entry<String, Map<String, List<String>>> o2) {
```

Comment:
Can we break this line? Does Java support trailing commas for that?
Also if the variables are renamed to `a` and `b`, we save two columns.

Reply:
Shortened.